src/code/z_kankyo.c

```diff
@@ -1,163 +1,992 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawCustomLensFlare.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Kankyo_InitGameOverLights.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_InitGameOverLights.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Kankyo_FadeInGameOverLights.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_FadeInGameOverLights.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Kankyo_FadeOutGameOverLights.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_FadeOutGameOverLights.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FC3DC.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_UpdateRain.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_FillScreen.s")
 
+u8 sSandstormPrimColors[] = {
+    255, 255, 255, //
+    255, 255, 255, //
+    255, 255, 255, //
+    255, 255, 255, //
+    210, 156, 85,  //
+    255, 200, 100, //
+    225, 160, 50,  //
+    105, 90,  40,  //
+};
+
+u8 sSandstormEnvColors[] = {
+    100, 100, 100, //
+    100, 100, 100, //
+    100, 100, 100, //
+    100, 100, 100, //
+    155, 106, 35,  //
+    200, 150, 50,  //
+    170, 110, 0,   //
+    50,  40,  0,   //
+};
+
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSandstorm.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD2B4.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_AdjustLights.s")
+
+void Environment_LerpRGB8(u8* from, Color_RGB8* to, f32 lerp, s16* dst) {
+    Color_RGB8 result;
+
+    Color_RGB8_Lerp((Color_RGB8*)from, to, lerp, &result);
+
+    dst[0] = result.r - from[0];
+    dst[1] = result.g - from[1];
+    dst[2] = result.b - from[2];
+}
+
+void Environment_LerpAmbientColor(PlayState* play, Color_RGB8* to, f32 lerp) {
+    Environment_LerpRGB8(play->envCtx.lightSettings.ambientColor, to, lerp, play->envCtx.adjLightSettings.ambientColor);
+}
+
+void Environment_LerpDiffuseColor(PlayState* play, Color_RGB8* to, f32 lerp) {
+    Environment_LerpRGB8(play->envCtx.lightSettings.light1Color, to, lerp, play->envCtx.adjLightSettings.light1Color);
+    Environment_LerpRGB8(play->envCtx.lightSettings.light2Color, to, lerp, play->envCtx.adjLightSettings.light2Color);
+}
+
+void Environment_LerpFogColor(PlayState* play, Color_RGB8* to, f32 lerp) {
+    Environment_LerpRGB8(play->envCtx.lightSettings.fogColor, to, lerp, play->envCtx.adjLightSettings.fogColor);
+}
+
+void Environment_LerpFog(PlayState* play, s16 fogNearTarget, s16 fogFarTarget, f32 lerp) {
+    play->envCtx.adjLightSettings.fogNear = (fogNearTarget - play->envCtx.lightSettings.fogNear) * lerp;
+    play->envCtx.adjLightSettings.zFar = (fogFarTarget - play->envCtx.lightSettings.zFar) * lerp;
+}
+
+// Repurposed from OoT to be more general
+u32 Environment_GetBgsDayCount(void) {
```

Comment:
What does "bgs" mean?

Reply:
Its from OoT. Based on usage I think it is `Biggoron Sword` but I think Engineer would know more.

Reply:
yes it is biggoron sword. in oot it keeps track of the days that have passed since you gave claim check, cause it takes 3 days for him to make the sword

if it gets used for other things in this game, a more general name could be thought of i suppose

Reply:
Yes exactly, it was just named after OoT early in WIP to help with cross-referencing the files since the function is identical, but in MM is repurposed to be more general. It was just never named to be more general but probably should be

Reply:
Posted in #mm-names-discussion